compiler/rustc_mir_transform/src/lib.rs

```diff
@@ -546,6 +547,7 @@ fn run_optimization_passes<'tcx>(tcx: TyCtxt<'tcx>, body: &mut Body<'tcx>) {
             &lower_slice_len::LowerSliceLenCalls, // has to be done before inlining, otherwise actual call will be almost always inlined. Also simple, so can just do first
             &unreachable_prop::UnreachablePropagation,
             &uninhabited_enum_branching::UninhabitedEnumBranching,
+            &simplify_pow_of_two::SimplifyPowOfTwo,
```

Comment:
This should probably go after const prop, to increase the likelihood to have constants.

Reply:
That's after inlining, with the current way it works there's a chance the inliner will inline the call, and thus we won't detect it.